be/src/olap/comparison_predicate.cpp

```diff
@@ -145,28 +146,68 @@ COMPARISON_PRED_COLUMN_BLOCK_EVALUATE(LessEqualPredicate, <=)
+                        sel[new_size] = idx;                                                       \
+                        const auto& cell_value =                                                   \
+                                reinterpret_cast<const vectorized::Int32&>(data_array[idx]);       \
+                        bool ret = !null_bitmap[idx] && (cell_value OP code);                      \
+                        new_size += _opposite ? !ret : ret;                                        \
+                    }                                                                              \
+                }                                                                                  \
+            } else {                                                                               \
+                auto* nested_col_ptr =                                                             \
+                        vectorized::check_and_get_column<vectorized::PredicateColumnType<type>>(   \
+                                nested_col);                                                       \
+                auto& data_array = nested_col_ptr->get_data();                                     \
+                for (uint16_t i = 0; i < *size; i++) {                                             \
+                    uint16_t idx = sel[i];                                                         \
+                    sel[new_size] = idx;                                                           \
+                    const type& cell_value = reinterpret_cast<const type&>(data_array[idx]);       \
+                    bool ret = !null_bitmap[idx] && (cell_value OP _value);                        \
+                    new_size += _opposite ? !ret : ret;                                            \
+                }                                                                                  \
             }                                                                                      \
             *size = new_size;                                                                      \
+        } else if (column.is_column_dictionary()) {                                                \
+            if constexpr (std::is_same_v<type, StringValue>) {                                     \
+                auto& dict_col =                                                                   \
+                        reinterpret_cast<vectorized::ColumnDictionary<vectorized::Int32>&>(column);\
+                auto& data_array = dict_col.get_data();                                            \
+                auto code = dict_col.find_code(_value);                                            \
+                if (code < 0 && IS_RANGE) {                                                        \
+                    code = dict_col.find_bound_code(_value, 0 OP 1, 1 OP 1);                       \
```

Comment:
If there is a range predicate here, we just need to use string sort rule to sort dict code.
This means ```data_array``` has the same order with string value.
Finally we can compare value and data_array directly here.
So why need ```find_bound_code``` here?

Reply:
For range comparison predicates, predicate constants in query sql may not be in the dictionary.

For example, the currently sorted dictionary is `a:0, c:1, e:2`, the predicate in sql is: `col > 'b' and col < 'd'`, `b` and `d` are not in the dictionary, but the result `c` needs to be returned , this is what `find_bound_code` does.